### PR TITLE
feat:  add `confrozenset()`

### LINKED ISSUE #1897: min_items not enforcable for FrozenSet
# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.6.1
            pydantic compiled: True
                 install path: /home/snazzy/code/test-project/venv/lib/python3.8/site-packages/pydantic
               python version: 3.8.2 (default, Jul 16 2020, 14:00:26)  [GCC 9.3.0]
                     platform: Linux-5.4.0-42-generic-x86_64-with-glibc2.29
     optional deps. installed: ['typing-extensions']
```

## Code Example

This works fine:

```py
from pydantic import BaseModel, Field
from typing import Set

class ModelOne(BaseModel):
    my_set: Set = Field(..., min_items=1)

this_should_fail = ModelOne(my_set=set())
```

Using `FrozenSet` instead of `Set` doesn't work:

```py
from pydantic import BaseModel, Field
from typing import FrozenSet

class ModelTwo(BaseModel):
    my_frozen_set: FrozenSet = Field(..., min_items=1)
```
This causes an error at definition time:
```
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "pydantic/main.py", line 252, in pydantic.main.ModelMetaclass.__new__
  File "pydantic/fields.py", line 308, in pydantic.fields.ModelField.infer
  File "pydantic/schema.py", line 864, in pydantic.schema.get_annotation_from_field_info
ValueError: On field "my_frozen_set" the following field constraints are set but not enforced: min_items. 
For more details see https://pydantic-docs.helpmanual.io/usage/schema/#unenforced-field-constraints
```

Since `FrozenSet` is simply the immutable version of `Set`, it should be possible to use `min_items` to validate the size of the set.

---
Edit: 

The reason I expected `FrozenSet` to work the same way as mutable `Set` is because [the docs](https://pydantic-docs.helpmanual.io/usage/types/#typing-iterables) describe them in the exact same manner. Even though frozen sets are not mentioned in the [constrained types](https://pydantic-docs.helpmanual.io/usage/types/#constrained-types) section, it's a little natural to (wrongly) assume parallel functionalities are available for it.

### PR DESCRIPTION
**Pull request by PrettyWood**
## Change Summary
Exactly the same as `conset()` but with `frozenset` to allow setting constraints on frozen sets

## Related issue number
fix #1897

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: pydantic/schema.py [hunk 1]
```diff
@@ -965,7 +966,9 @@ def go(type_: Any) -> Type[Any]:
                 used_constraints.update({'min_items', 'max_items'})
                 return conlist(go(args[0]), min_items=field_info.min_items, max_items=field_info.max_items)
 
-            if issubclass(origin, Set) and (field_info.min_items is not None or field_info.max_items is not None):
+            if issubclass(origin, AbstractSet) and (
```
Comment (slorg1) [inline pydantic/schema.py:969]: you seem to "remove" the difference between the treatment of `Set` and `FrozenSet` in this PR, have you seen this guy?
https://github.com/samuelcolvin/pydantic/blob/5261fd05a0374b84ce2602d45990baf480fa2417/pydantic/fields.py#L589
It looks like they are treated differently there and your PR does not touch it (idk if it is on purpose)

If I follow your logic, you'd want the behaviour to be consistent between `Set` and `Frozenset`. It is unclear to me why the validators would (optionally) apply to 1 and not in the other.

Forgive me if this does not make sense, I am new to pydantic and I do not yet know all the ins and outs.

Comment (PrettyWood) [reply pydantic/schema.py:969]: We are just talking about a quick fix for schema generation. The goal is not to go over all the validation and internals

File: tests/test_types.py [hunk 2]
```diff
@@ -488,6 +488,36 @@ class Model(BaseModel):
...
+        {
+            'loc': ('x',),
+            'msg': 'ensure this value has at least 1 items',
+            'type': 'value_error.set.min_items',
+            'ctx': {'limit_value': 1},
+        }
+    ]
+
+    Model(x=frozenset({'a'}))
+    Model(x=frozenset({'a', 'b', 'c'}))
```
Comment (samuelcolvin) [inline tests/test_types.py:507]: can we check that `.dict()` is correct.

Can we also try with a list or normal set or something?

**Review by samuelcolvin — COMMENTED** [review]
otherwise LGTM.

**Comment by samuelcolvin** [comment]
please update.

**Comment by PrettyWood** [comment]
Thanks for the review. I went too fast on this one since I used to call `conset` for `Set` or `FrozenSet`.
I hence added `confrozenset`, which is a copy paste of `conset`. Please review!

PS: the second commit is just some work to add tests for `allowed_schemes` and I went a bit further by allowing `Collection`.
I feel like in _pydantic_ "input types" tend to be a bit strict. If you don't agree with this change or want it to be in another PR I can rebase and remove it

**Comment by samuelcolvin** [comment]
thank you.

**Pull request merged**